recycle-node.rb

```diff
@@ -80,19 +80,19 @@ def node_ready?(node)
 
 def worker_node?(node)
   node.dig("metadata", "labels")["kubernetes.io/role"] == "node" \
-    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup") == WORKER_NODE_INSTANCEGROUP
+    && node.dig("metadata", "labels", "kops.k8s.io/instancegroup").include?(WORKER_NODE_INSTANCEGROUP)
 end
 
 def get_worker_instance_group_size
   docs = []
   YAML.load_stream(get_kops_config) { |doc| docs << doc }
-  worker_instance_group = docs.last
+  worker_instance_group = docs.last(3)
```

Comment:
Per my comment above, can we identify the worker node instance groups via a label?

If we can do that, then we don't need to worry about their position in the YAML file (i.e. `last(3)`). This is less brittle, so we won't break stuff if we reorder the sections of the yaml file.